test/Semantic/Spec.hs

```diff
@@ -24,5 +25,22 @@ spec = parallel $ do
     it "renders with the specified renderer" $ do
       output <- fmap runBuilder . runTaskOrDie $ parseTermBuilder TermSExpression [methodsBlob]
       output `shouldBe` "(Statements\n  (Method\n    (Empty)\n    (Identifier)\n    (Statements)))\n"
+
+  describe "git ls-tree parsing" $ do
+    it "parses a git output string" $ do
+      let input = "100644 tree abcdef\t/this/is/the/path"
+      let expected = TreeEntry NormalMode TreeObject (OID "abcdef") "/this/is/the/path"
+      parseEntry input `shouldBe` expected
+
+    it "allows whitespace in the path" $ do
+      let input = "100644 tree 12345\t/this\n/is\t/the /path\r"
+      let expected = TreeEntry NormalMode TreeObject (OID "12345") "/this\n/is\t/the /path\r"
+      parseEntry input `shouldBe` expected
+
+    it "parses many outputs separated by \\NUL" $ do
+      let input = "100644 tree abcdef\t/this/is/the/path\NUL120000 blob 17776\t/dev/urandom\NUL\n"
+      let expected = [ TreeEntry NormalMode TreeObject (OID "abcdef") "/this/is/the/path", TreeEntry SymlinkMode BlobObject (OID "17776") "/dev/urandom"]
```

Comment:
Is this a 17776 reference (by Jon Bois)? Love it.

Reply:
Yes, it is.